plugins/sbt/src/index.ts

```diff
@@ -0,0 +1,198 @@
+import {
+  Auto,
+  execPromise,
+  getCurrentBranch,
+  IPlugin,
+  validatePluginConfiguration,
+} from "@auto-it/core";
+import { inc, ReleaseType } from "semver";
+import * as t from "io-ts";
+import stripAnsi from "strip-ansi";
+
+const pluginOptions = t.partial({
+  manageVersion: t.boolean,
+});
+
+export type ISbtPluginOptions = t.TypeOf<typeof pluginOptions>;
+
+/** Publish Scala projects with sbt */
+export default class SbtPlugin implements IPlugin {
+  /** The name of the plugin */
+  name = "sbt";
+
+  /** The options of the plugin */
+  readonly options: ISbtPluginOptions;
+
+  /** Initialize the plugin with it's options */
+  constructor(options: ISbtPluginOptions) {
+    this.options = options;
+  }
+
+  /** Tap into auto plugin points. */
+  apply(auto: Auto) {
+    // exact copy-paste from the git-tag plugin
+    /** Get the latest tag in the repo, if none then the first commit */
+    async function getTag() {
+      try {
+        return await auto.git!.getLatestTagInBranch();
+      } catch (error) {
+        return auto.prefixRelease("0.0.0");
+      }
+    }
+
+    /** Calls sbt in the client and returns cleaned up logs */
+    async function sbtClient(...args: string[]) {
+      const output = await execPromise("sbt", ["--client", ...args]);
+      const cleanOutput = stripAnsi(output).replace(/(.*\n)*^>.*$/m, "").trim();
+      return cleanOutput;
+    }
+
+    /** Read version from sbt */
+    async function sbtGetVersion() {
+      // in multi-module projects, we want to get only ThisBuild/version
+      await sbtClient("set version/aggregate := false");
+      const output = await sbtClient("print version");
+      const version = output.split("\n").shift()?.trim();
+      if (!version) {
+        throw new Error(`Failed to read version from sbt: ${output}`);
+      }
+
+      auto.logger.log.info(`Got version from sbt: ${version}`);
+      return version;
+    }
+
+    /** Set version in sbt to the given value */
+    async function sbtSetVersion(version: string) {
+      auto.logger.log.info(`Set version in sbt to "${version}"`);
```

Comment:
You'll want to make these logs use the `verbose` logger instead of the default `log`.

Reply:
Done in 8c2dda9